### ISSUE TITLE
Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead.

### ISSUE THREAD

**Issue by helgatheviking** [woothemes/woocommerce#46879]
> [!IMPORTANT]
> This issue aims to apply the necessary changes in WooCommerce Admin. The necessary changes for WooCommerce Blocks can be found in #48796, #48797 and #48801.

### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Getting the `Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead.` warning notice in the console from at least the two following compiled assets:

```
woocommerce/assets/client/blocks/cart-frontend.js
woocommerce/assets/client/admin/app/index.js
```

At least the former is traced back to the cart's [`render-frontend.jsx` file](https://github.com/woocommerce/woocommerce/blob/c5e07b8dd2c6030c9098f643f64901f799e9565d/plugins/woocommerce-blocks/assets/js/base/utils/render-frontend.tsx#L4) - hat tip to @senadir 

It's unclear to me where the other one is coming from as I find it reallllly hard to track these files back to their sources.

The solution is to import both `render` and `createRoot` and then conditionally use `createRoot` if it exists... which it should now for the past 2 versions of WP. I would send a PR, but as an outside contributor the monorepo is becoming almost unnavigable to me (especially on the Javascript side)

```
import { createRoot, render } from '@wordpress/element';

if ( createRoot ) {
	createRoot( target ).render( <App /> );
} else {
	render( <App />, target );
}
```

### Expected behavior

No warnings

### Actual behavior

Warnings in the console

### Steps to reproduce

Go to the page with the cart block
OR
Go to the orders admin screen, reports, or customers, etc

### WordPress Environment

Repo version of Woo at eb75902c4cb5001e94db69eac6d8bf7c7b790b5f

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
```[tasklist]
### Tasks
- [ ] https://github.com/woocommerce/woocommerce/pull/48831
- [ ] https://github.com/woocommerce/woocommerce/pull/48832
- [ ] https://github.com/woocommerce/woocommerce/pull/48833
- [ ] https://github.com/woocommerce/woocommerce/pull/48834
- [ ] https://github.com/woocommerce/woocommerce/pull/48835
- [ ] https://github.com/woocommerce/woocommerce/pull/48785
- [ ] https://github.com/woocommerce/woocommerce/issues/48796
- [ ] https://github.com/woocommerce/woocommerce/issues/48797
- [ ] https://github.com/woocommerce/woocommerce/issues/48801
- [ ] https://github.com/woocommerce/woocommerce/pull/50765
```

**Comment by rjchow** [comment]
I did an AST search for render from '@wordpress/element', and it looks like there will be a few places where we need to work on. 

I'll work on PRs remedying all of them under WC Core, and assign the PRs to be completed by relevant teams - the code change is trivial but it'll be too much work for me to test and especially write testing instructions.

![image](https://github.com/woocommerce/woocommerce/assets/27843274/18bf322d-be48-46bb-a10b-8ae1968a08e3)

**Comment by samueljseay** [comment]
Thanks for raising this @helgatheviking 

This should be fixed in development now that https://github.com/woocommerce/woocommerce/pull/48843 is merged.

You also should never see these warnings in production because React is designed to only show them in a development environment.

**Comment by helgatheviking** [comment]
I am developing with woo all day everyday day so it was very annoying. Happy to see this resolved! Good work y'all!

**Comment by nielslange** [comment]
> Thanks for raising this @helgatheviking
> This should be fixed in development now that https://github.com/woocommerce/woocommerce/pull/48843 is merged.

@samueljseay and @helgatheviking The issue is fixed for the WooCommerce Blocks, but might still appear for other sections, such as WooCommerce Admin. I assume that this issue here had been created with WooCommerce Blocks in mind. As the issue affected various areas, @rjchow started creating PRs against it. To be able to see only areas that affect WooCommerce Blocks, I created the following issues, which have all been addressed:

- https://github.com/woocommerce/woocommerce/issues/48796
- https://github.com/woocommerce/woocommerce/issues/48797
- https://github.com/woocommerce/woocommerce/issues/48801

So, to clarify, while this particular issue is still open, the problem within WooCommerce Blocks has been addressed.
